deepr/jobs/export_xla_model_metadata.py

```diff
@@ -0,0 +1,100 @@
+"""Export xla compatible model metadata from a saved model"""
+
+from dataclasses import dataclass
+import logging
+from typing import List, Dict
+
+import tensorflow as tf
+
+from deepr.io import Path
+from deepr.jobs import base
+from deepr.utils import import_graph_def
+from deepr.jobs._xla_generated.tf2xla_pb2 import Config as XlaMetaConfig
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ExportXlaModelMetadata(base.Job):
+    """Export xla compatible model metadata from a saved model
+
+    Attributes
+    ----------
+    path_saved_model : str
+        Path to directory containing SavedModel exports to convert
+    path_metadata : str
+        Path to directory that will contain the metadata
+    graph_name : str
+        Name of the saved model graph (name of the protobuf file)
+    metadata_name : str
+        Name of the metadata file
+    feed_shapes : Dict[str, List[int]]
+        Shapes of feeds to expose
+    fetch_shapes : Dict[str, List[int]]
+        Shapes of fetches to expose
+    """
+
+    path_saved_model: str
+    path_metadata: str
+    graph_name: str
+    metadata_name: str
+    feed_shapes: Dict[str, List[int]]
+    fetch_shapes: Dict[str, List[int]]
+
+    def run(self):
+        # Create session and import graph under scope "model"
+        session = tf.Session(graph=tf.Graph())
+        with session.graph.as_default():
+            import_graph_def(f"{self.path_saved_model}/{self.graph_name}", name="")
+
+            feed_nodes = {
+                "/".join(name.split("/")): node
+                for name, node in get_nodes(
+                    session.graph_def, [f"{name}" for name in self.feed_shapes.keys()]
+                ).items()
+            }
+            fetch_nodes = {
+                "/".join(name.split("/")): node
+                for name, node in get_nodes(
+                    session.graph_def, [f"{name}" for name in self.fetch_shapes.keys()]
+                ).items()
+            }
+            meta = XlaMetaConfig()
+
+            for feed_name in feed_nodes:
```

Comment:
suggestion: `for name, node in feed_nodes.items()`

Reply:
done